nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -78,10 +78,10 @@
 
     protected String countersCollectionName;
 
-    public MongoDBSession(MongoDBDirectory directory) {
+    public MongoDBSession(MongoDBDirectory directory, MongoClient client) {
         super(directory, MongoDBReference.class);
         MongoDBDirectoryDescriptor desc = directory.getDescriptor();
-        client = MongoDBConnectionHelper.newMongoClient(desc.getServerUrl());
+        this.client = client;
```

Comment:
I'd prefer a `getClient` protected method on the MongoDBDirectory, and have the MongoDBSession constructor just do `client = directory.getClient()`.